cmd/api/book/service.go

```diff
@@ -101,9 +105,12 @@ type ListBooksRequest struct {
 	PageSize      int
 }
 
-func (s *Service) ListBooks(params ListBooksRequest) (PagedBooks, error) {
-	itemsTotal, err := s.repo.ListBooksTotals(params.Name, params.MinPrice, params.MaxPrice, params.Archived)
+func (s *Service) ListBooks(ctx context.Context, params ListBooksRequest) (PagedBooks, error) {
+	itemsTotal, err := s.repo.ListBooksTotals(ctx, params.Name, params.MinPrice, params.MaxPrice, params.Archived)
 	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			return PagedBooks{}, err
```

Comment:
Please wrap the error.

Reply:
Which new information could I add here?
It because the error arrives already wrapped at this point: 

>...
> row := store.db.QueryRowContext(ctx, sqlStatement, name, minPrice32, maxPrice32, archived)
> 	var count int
> 	err := row.Scan(&count)
> 	if err != nil {
> 		return count, fmt.Errorf("counting books from db: %w", err)
> 	}
> ...

Reply:
Watch which error it returned when I was testing the error route: (timeout set to 5mS to force ctx to timeout):

![image](https://github.com/cesarFuhr/books-service/assets/95890275/d57b51e0-d81d-4ad6-b0cc-acd686ec2f61)

Reply:
Just to talk on next call...

Reply:
And we need to talk about the log.Prints too. Just to remember...

Reply:
This is not correct, yeah... we need to talk about it.

Reply:
> Which new information could I add here? It because the error arrives already wrapped at this point:
> 
> > ...
> > row := store.db.QueryRowContext(ctx, sqlStatement, name, minPrice32, maxPrice32, archived)
> > var count int
> > err := row.Scan(&count)
> > if err != nil {
> > return count, fmt.Errorf("counting books from db: %w", err)
> > }
> > ...

Its already wrapped in the database layer.
You should add some context in the service layer. I now it may sound repetitive, but it gives you a nice trace on where the error passed.